pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -74,30 +73,39 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 		return nil
 	}
 
-	// we can always allow the upgrade if there isn't a version already installed
-	if len(cv.Status.History) == 0 {
-		klog.V(2).Infof("Precondition %s passed: no release history.", pf.Name())
-		return nil
+	currentVersion, err := semver.Parse(cv.Status.Desired.Version)
+	if err != nil {
+		return &precondition.Error{
+			Nested:             err,
+			Reason:             "InvalidCurrentVersion",
+			Message:            err.Error(),
+			Name:               pf.Name(),
+			NonBlockingWarning: true, // do not block on issues that require an update to fix
+		}
 	}
 
-	currentVersion := GetCurrentVersion(cv.Status.History)
-	currentMinor := GetEffectiveMinor(currentVersion)
-	desiredMinor := GetEffectiveMinor(releaseContext.DesiredVersion)
-	klog.V(2).Infof("currentMinor %s releaseContext.DesiredVersion %s desiredMinor %s", currentMinor, releaseContext.DesiredVersion, desiredMinor)
+	targetVersion, err := semver.Parse(releaseContext.DesiredVersion)
+	if err != nil {
+		return &precondition.Error{
+			Nested:  err,
+			Reason:  "InvalidDesiredVersion",
+			Message: err.Error(),
+			Name:    pf.Name(),
+		}
+	}
 
-	// if there is no difference in the minor version (4.y.z where 4.y is the same for current and desired), then we can still upgrade
-	// if no cluster overrides have been set
-	if !minorVersionUpgrade(currentMinor, desiredMinor) {
-		klog.V(2).Infof("Precondition %q passed: minor from the target %s is not a minor version update from the current %s.%s.", pf.Name(), releaseContext.DesiredVersion, currentVersion, currentMinor)
+	klog.V(2).Infof("The current version is %s parsed from %s and the target version is %s parsed from %s", currentVersion.String(), cv.Status.Desired.Version, targetVersion.String(), releaseContext.DesiredVersion)
```

Comment:
nit: this "we parsed the versions!" log doesn't seem like `V(2)`.  If you want to keep it, can you move it to `V(4)` or deeper?  I see there was a `klog.V(2)` chattering about `releaseContext.DesiredVersion` earlier, which is quite possibly me making that same decision, but no reason we can't do better now :)

Logs for "I'm doing (or just did) `$ACTION` in `$CONTEXT`" are more interesting than the ones that are more "here's a context dump..." (and parsing internal data is small enough that I don't think it qualifies as an action for admins reading logs).